app/controllers/api/v1/study_files_controller.rb

```diff
@@ -293,11 +293,18 @@ def format_log_props(status, error_message)
 
       # update the given study file with the request params and save it
       # raises ArgumentError if the save fails due to model validations
-      # if the save is successful and includes a complete upload, will send it to firecloud
+      # if the save is successful and includes a complete upload, will send it to Terra
       # and kick off a parse job if requested
-      def perform_update(study_file)
+      def perform_update!(study_file)
         study = study_file.study
-        safe_file_params = study_file_params
+        if study_file_params[:file_type] == 'AnnData'
+          study_file.build_ann_data_file_info if study_file.ann_data_file_info.blank?
```

Comment:
I just realized this, sorry I hadn't raised it sooner.

Rendering "AnnData" as `ann_data` and not `anndata` strikes me as counterintuitive to most developers who work with AnnData, and different from the preponderance of the rest of SCP code (even within SCP Ruby).  That's a foreseeable source of bugs and development snags. 

While I can imagine valid theoretical arguments for this variable naming, I can't find any empirical examples outside some (seemingly inconsistent) places in this Ruby codebase where we write "AnnData" as "ann_data" and not "anndata".  Beyond that limited case, I only see examples of "anndata":

* `reference_anndata_file` later in this file
* Ingest Pipeline uniformly uses "[anndata](https://github.com/search?q=repo%3Abroadinstitute%2Fscp-ingest-pipeline%20anndata&type=code)"
* A close corrolary: the Ingest Pipeline PAPI arguments use `--ingest-anndata --anndata-file`
* Places we log AnnData-related things to Mixpanel, e.g. `anndata-upload-experience`

I notice a fair amount of existing internal inconsistency in our .rb files on this -- case-sensitive search shows 65 results for "anndata", 50 for "ann_data".  This is a trivial thing, in a way, but [naming things is a hard problem](https://martinfowler.com/bliki/TwoHardThings.html), and I think consistently naming here would reduce errors and development roadbumps.

Are there easy ways we can modestly decrease this naming consistency in this PR?   Or would a partial solution just exacerbate the likelihood of bugs and snags?

Reply:
Thanks for chatting.  Given this would require monkey-patching some Ruby / Rails built-in code, the effort seems not clearly worth the benefit of increasing global name consistency.